p8e-util/src/main/kotlin/io/p8e/crypto/SignatureInputStream.kt

```diff
@@ -2,63 +2,67 @@ package io.p8e.crypto
 
 import java.io.InputStream
 import java.security.PublicKey
+import java.security.Signature
 
 class SignatureInputStream(
     private val inputStream: InputStream,
-    private val signer: SignerImpl,
+    private val signer: SignerImpl? = null,
```

Comment:
Probably not vital, but since we only ever want either a signer, or a verifier, we could probably make this an enum class with two entries to wrap the signer/verifier and only have one argument and then match based on the type, to make that requirement more clear?

Reply:
Yea, i can get this cleaned up, was too excited to get this to TEST and see if it fixes :allthethings 😅

Reply:
Haha, yeah, don't want to slow that process down 😆